### PR TITLE
Add support for WebGL extensions EXT_clip_control, EXT_depth_clamp, ...

### PR DESCRIPTION
**Pull request by juj**
Add support for WebGL extensions EXT_clip_control, EXT_depth_clamp, EXT_polygon_offset_clamp and WEBGL_polygon_mode.

### CODE REVIEW

File: system/include/webgl/webgl1_ext.h [hunk 1]
```diff
@@ -370,9 +469,86 @@ WEBGL_APICALL void GL_APIENTRY glMultiDrawElementsInstancedWEBGL(GLenum mode,
...
+#define GL_UPPER_LEFT_EXT          0x8CA2
+#define GL_NEGATIVE_ONE_TO_ONE_EXT 0x935E
+#define GL_ZERO_TO_ONE_EXT         0x935F
+#define GL_CLIP_ORIGIN_EXT         0x935C
+#define GL_CLIP_DEPTH_MODE_EXT     0x935D
+WEBGL_APICALL void GL_APIENTRY emscripten_glClipControlEXT(GLenum origin, GLenum depth);
+WEBGL_APICALL void GL_APIENTRY glClipControlEXT(GLenum origin, GLenum depth);
+#endif
+
+// 52. https://registry.khronos.org/webgl/extensions/EXT_depth_clamp/
```
Comment (sbc100) [inline system/include/webgl/webgl1_ext.h:521]: Dies these numbers have some meaning?

File: test/browser/webgl2_simple_enable_extensions.c [hunk 2]
```diff
@@ -61,6 +61,15 @@ int main()
...
     assert(emscripten_webgl_enable_WEBGL_multi_draw_instanced_base_vertex_base_instance(context));
 
+  if (hasext(exts, "EXT_polygon_offset_clamp"))
+    assert(emscripten_webgl_enable_EXT_polygon_offset_clamp(context));
+
+  if (hasext(exts, "EXT_clip_control"))
+    assert(emscripten_webgl_enable_EXT_clip_control(context));
+
+  if (hasext(exts, "WEBGL_polygon_mode"))
+    assert(emscripten_webgl_enable_WEBGL_polygon_mode(context));
```
Comment (sbc100) [inline test/browser/webgl2_simple_enable_extensions.c:71]: It lots of places in this PR I see only 3 extensions (looks like `EXT_depth_clamp` is missing here and elsewhere).  Is there some reason for that?

**Review by sbc100 — COMMENTED** [review]
lgtm % comments

File: system/include/webgl/webgl1_ext.h (hunk 1 above)
Comment (kainino0x) [reply system/include/webgl/webgl1_ext.h:521]: Yes, they're part of the extension specification.

> ## Number
> WebGL extension \#52

Comment (kainino0x) [reply system/include/webgl/webgl1_ext.h:521]: (oops didn't finish writing)
It's not a very important number but it's a fixed number and I think it's in the order that the extensions were drafted. https://registry.khronos.org/webgl/extensions/

**Review by kainino0x — APPROVED** [review]

**Comment by juj** [comment]
> lgtm % comments

@sbc100 Did Kai's comments resolve your questions, or were there still pending items?

**Pull request merged**